airbyte-integrations/connectors/source-faker/source_faker/streams.py

```diff
@@ -89,73 +104,79 @@ def state(self, value: Mapping[str, Any]):
         self._state = value
 
     def generate_user(self, user_id: int):
-        time_a = self.dt.datetime()
-        time_b = self.dt.datetime()
+        time_a = dt.datetime()
+        time_b = dt.datetime()
 
         profile = {
             "id": user_id + 1,
             "created_at": format_airbyte_time(time_a if time_a <= time_b else time_b),
             "updated_at": format_airbyte_time(time_a if time_a > time_b else time_b),
-            "name": self.person.name(),
-            "title": self.person.title(),
-            "age": self.person.age(),
-            "email": self.person.email(),
-            "telephone": self.person.telephone(),
-            "gender": self.person.gender(),
-            "language": self.person.language(),
-            "academic_degree": self.person.academic_degree(),
-            "nationality": self.person.nationality(),
-            "occupation": self.person.occupation(),
-            "height": self.person.height(),
-            "blood_type": self.person.blood_type(),
-            "weight": self.person.weight(),
+            "name": person.name(),
+            "title": person.title(),
+            "age": person.age(),
+            "email": person.email(),
+            "telephone": person.telephone(),
+            "gender": person.gender(),
+            "language": person.language(),
+            "academic_degree": person.academic_degree(),
+            "nationality": person.nationality(),
+            "occupation": person.occupation(),
+            "height": person.height(),
+            "blood_type": person.blood_type(),
+            "weight": person.weight(),
         }
 
         while not profile["created_at"]:
-            profile["created_at"] = format_airbyte_time(self.dt.datetime())
+            profile["created_at"] = format_airbyte_time(dt.datetime())
 
         if not profile["updated_at"]:
             profile["updated_at"] = profile["created_at"] + 1
 
-        return profile
+        record = AirbyteRecordMessage(stream=self.name, data=profile, emitted_at=now_millis())
+        return AirbyteMessageWithCachedJSON(type=Type.RECORD, record=record)
 
     def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
         total_records = self.state[self.cursor_field] if self.cursor_field in self.state else 0
         records_in_sync = 0
-        records_in_slice = 0
 
         median_record_byte_size = 450
         yield generate_estimate(self.name, self.count - total_records, median_record_byte_size)
 
-        for i in range(total_records, self.count):
-            user = self.generate_user(i)
-            yield user
-            total_records += 1
-            records_in_sync += 1
-            records_in_slice += 1
+        running = True
```

Comment:
is there value in the `running` var? seems like wherever we set it to false we also `break`